extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/health/KafkaHealthCheck.java

```diff
@@ -3,36 +3,39 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import javax.annotation.PostConstruct;
 import javax.annotation.PreDestroy;
 import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
 
 import org.apache.kafka.clients.admin.AdminClient;
 import org.apache.kafka.clients.admin.AdminClientConfig;
 import org.apache.kafka.common.Node;
-import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.health.HealthCheck;
 import org.eclipse.microprofile.health.HealthCheckResponse;
 import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
 import org.eclipse.microprofile.health.Readiness;
 
+import io.quarkus.kafka.client.runtime.KafkaRuntimeConfig;
+
 @Readiness
 @ApplicationScoped
 public class KafkaHealthCheck implements HealthCheck {
 
-    @ConfigProperty(name = "kafka.bootstrap.servers", defaultValue = "localhost:9092")
-    private String bootstrapServers;
-
     private AdminClient client;
 
-    @PostConstruct
-    void init() {
-        Map<String, Object> conf = new HashMap<>();
-        conf.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    @Inject
+    public KafkaHealthCheck(KafkaRuntimeConfig runtimeConfig) {
+        Map<String, Object> conf = new HashMap<>(runtimeConfig.getProperties());
```

Comment:
How is this related to the issue? I may miss a link.

Reply:
the previous `@PostConstruct` ran into NPEs for the `@Inject`ed `KafkaRuntimeConfig`. Using constructor injection it worked.

Reply:
Why not using the static access to the configuration? That would remove the bean.